hardware_interface/src/resource_manager.cpp

```diff
@@ -1283,6 +1304,58 @@ HardwareReadWriteStatus ResourceManager::read(
   return read_write_status;
 }
 
+void ResourceManager::publish_all_interface_values() const
+{
+  control_msgs::msg::DynamicInterfaceValues interface_values;
+  interface_values.header.stamp = resource_storage_->clock_interface_->get_clock()->now();
+
+  control_msgs::msg::InterfaceValue state_interface_values;
+  for (const auto & state_interface_name : resource_storage_->available_state_interfaces_)
+  {
+    try
+    {
+      const auto state_interface_value =
+        resource_storage_->state_interface_map_.at(state_interface_name).get_value();
+      state_interface_values.interface_names.push_back(state_interface_name);
+      state_interface_values.values.push_back(state_interface_value);
+    }
+    catch (const std::out_of_range & e)
+    {
+      RCUTILS_LOG_WARN_NAMED(
```

Comment:
Out of curiosity, do you think this should be throttled? or do you think it is fine to have like this?

Reply:
i am not sure about the performance penalty, but since this case should not happen under normal circumstances, i think its fine like this. But any thoughts/corrections are welcome.